server/project/project.go

```diff
@@ -228,6 +244,12 @@ func (s *Server) Update(ctx context.Context, q *ProjectUpdateRequest) (*v1alpha1
 		return nil, err
 	}
 
+	for _, groupKind := range uniqueGroupKinds(append(q.Project.Spec.ClusterResourceWhitelist, oldProj.Spec.ClusterResourceWhitelist...)) {
+		if err := s.enf.EnforceErr(ctx.Value("claims"), rbacpolicy.ResourceClusters, rbacpolicy.ActionUpdate, groupKind); err != nil {
```

Comment:
instead of supplying groupkind to EnforceErr, we need to iterate all clusters listed in the project, and call EnforceErr on each cluster address, to verify they have `update` privileges to that cluster.

Reply:
I've got this totally wrong :-(